Public/Src/Tools/Tool.MsBuildGraphBuilder/MsBuildGraphBuilder.cs

```diff
@@ -182,7 +186,11 @@ public static class MsBuildGraphBuilder
                         locatedMsBuildPath);
                 }
 
-                return ProjectGraphWithPredictionsResult.CreateSuccessfulGraph(projectGraphWithPredictions, assemblyPathsToLoad, locatedMsBuildPath);
+                return ProjectGraphWithPredictionsResult.CreateSuccessfulGraph(
+                    projectGraphWithPredictions, 
+                    assemblyPathsToLoad, 
+                    locatedMsBuildPath, 
+                    environmentVariablesLogger.PotentialEnvironmentVariablesReads.OrderBy(s => s));

```

Comment:
>PotentialEnvironmentVariablesReads [](start = 47, length = 34)

how does the logger inform that the read env variables are not available?

Reply:
Never mind. Saw that. But in that case I suggest we not only store the enumeration, but something like a Failure<> or Maybe<> to represent that the prediction is not there, and ideally a reason (like which resolver is to blame)

---
In reply to: [306449763](https://github.com/microsoft/BuildXL/pull/648#discussion_r306449763) [](ancestors = 306449763)

Reply:
Changes made in next update. The PropertyTrackingLogger will now surface a Possible<>. If it returns a failure, it'll be reported.